sdk/python/feast/infra/provider.py

```diff
@@ -238,9 +238,11 @@ def _get_column_names(
             reverse_field_mapping[col] if col in reverse_field_mapping.keys() else col
             for col in feature_names
         ]
+    _feature_names = set(feature_names) - set(join_keys)
```

Comment:
Nitpick

I think it would be helpful if you
1. Added a comment above this code section to describe what you are doing
2. Perhaps have the `list(_feature_names)` outside of the return in order to encapsulate the logic a bit more cleanly

Reply:
Sure done